src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Networking/INetworkConnection.cs

```diff
@@ -30,8 +30,10 @@ public interface INetworkConnection
         /// <summary>
         /// Call to send data to the connected peer
         /// </summary>
-        /// <param name="data">a reference to the data to send, data will be set to null after a send/the INetworkConnection will take over ownership of the array</param>
-        void Send(ref byte[] data);
+        /// <param name="data">A reference to the data to send</param>
+        /// <param name="offset">The offset from the start of the array to use to obtain the data to send</param>
+        /// <param name="length">The length of the data to send</param>
+        void Send(byte[] data, long offset, long length);
```

Comment:
As far as I understood, Span seems the way to go for memory spans and is available in Unity with a NuGet package. So why not Span? I am sure this has been discussed but it would be good to have the decision and rationale in writing.

Reply:
The issue is that support for consuming nuget packages in unity won't arrive until after the networking abstraction ships.
Options discussed included copy/pasting part of the nuget package. But the simplest is to use this interface and accept a minor breaking change when span becomes available. (byte[],int,int) will be fairly painless to migrate to span, (ref byte[]) less so.